### PR TITLE
Fail loudly

### PR DESCRIPTION
**Pull request by peterdemartini**
A better way of handling errors

### CODE REVIEW

**Comment by peterdemartini** [comment]
See #10

File: cmds/run.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ exports.handler = (argv) => {
     tjmFunctions.alreadyRegisteredCheck(jobContents)
         .catch((err) => {
             if (err) {
-                reply.warning(`Job ${jobContents.tjm.job_id} is already registered with cluster ${argv.c}`);
+                reply.warning(err);
             }
             return tjmFunctions.teraslice.jobs.submit(jobContents);
```
Comment (ciorg) [inline cmds/run.js]: I should have been more specific.  The desired behavior is that if the job has not been run before then it would register the job on the cluster, add the tjm data to the file, and run the job.  This commit breaks that work flow and new/ unregistered jobs end with the error 'Cannot read property 'id' of undefined'.

File: cmds/cmd_functions/functions.js [hunk 2]
```diff
@@ -18,14 +24,14 @@ module.exports = (argv, clusterName) => {
...
-                        return Promise.resolve(true);
+                        return Promise.resolve();
                     }
-                    return Promise.resolve(false);
+                    return Promise.reject(new Error('Job is not on the cluster'));
                 });
         }
-        return Promise.resolve(false);
+        return Promise.reject(new Error('No cluster configuration for this job'));
     }
```
Comment (ciorg) [inline cmds/cmd_functions/functions.js]: Actually now that I think of it, I don't think this should return an error if the job isn't on the cluster just in cases where the job is new then it would just be some different logic.  Makes it easier instead of having to run register for every new job, the jobs would be automatically registered and then continue.  This may only apply to run.

File: cmds/run.js [hunk 3]
```diff
@@ -22,14 +25,14 @@ exports.handler = (argv) => {
...
+        .catch((err) => {
+            if (err) {
+                reply.warning(err);
             }
-            return tjmFunctions.teraslice.jobs.submit(jobContents)
+            return tjmFunctions.teraslice.jobs.submit(jobContents);
         })
+        .then(() => tjmFunctions.loadAsset())
         .then((jobResult) => {
             const jobId = jobResult.id();
```
Comment (ciorg) [inline cmds/run.js]: Seeing the error 'Cannot read property 'id' of undefined' because it's trying to read a job id from the loadAsset function.   Should be reconfigured to register the job if necessary and return the needed job data to either update the job file and run the job or just run the job. Seems like the loadAsset should be at the beginning of the function. (if you are pressed for time with other things, I could finish this up.  Everything else looks really good and is a huge improvement)

**Comment by ciorg** [comment]
tl;dr the current PR breaks the run and register commands, I want to merge your fork onto a branch in this repo so that I can fix the register/ run command and then merge into master.  Which would also set a president for how work is done on this project going forward.

The run and register commands relied on the alreadyRegisteredCheck to return a true or false value to function properly, so as it is neither work and I can't merge the pull request without the functionality to handle new jobs or jobs that don't have tjm data.   
But, I'm also thinking now that the run command is basically redundant because the start command starts jobs that have tjm data and a flag could easily be added to the register command to start after they are registered with the cluster.
*tjm register -r jobFile.json -c localhost:5678* where the -r flag would indicate that the job would be run immediately.  With this change the run command could just be removed.
I took the liberty of adding the flag option to the register command, but because of the weirdness of the forking situation I can't just commit it to your branch.  
So maybe the best thing to do would be to merge your fork to a branch in this repo and I could change the register command on that branch and also add tests for all the commands before it is merged into master.  Kimbro would need to grant me permission to do this as it is I can't add branches.

**Comment by peterdemartini** [comment]
It felt odd to return a boolean without a reason, especially since there were multiple reasons why it return false. I’d be okay with returning more context that a boolean. 

I also agree with you that some of these commands might be conflated. Let’s have a discussion about that after we fix get this merged.

**Comment by ciorg** [comment]
Going to close this pull request because it is covered in pull request #23
